File: Python/Linked List/Merge Nodes in Between Zeros.py
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        newNode = ListNode(0)
        node = newNode
        curr=head.next
        while curr !=None:
            sum=0
            while curr.val!=0:
                sum=sum+curr.val
                curr=curr.next
            curr=curr.next
            temp=ListNode(sum)
            node.next=temp
            node=node.next
        result = newNode.next
        return result

File: CPP/Linked List/Merge Nodes in Between Zeros.cpp
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        // ListNode* node = head->next;
        // ListNode* merge=node;
        // while(merge!=NULL){
        //     int sum=0;
        //     while(merge->val!=0){
        //         sum=sum+merge->val;
        //         merge = merge->next;
        //     }
        //     node->val=sum;
        //     merge=merge->next;
        //     node->next=merge;
        //     node = node->next;

        // }
        // return head->next;
    //--------Another Approach---------------
        ListNode* newNode = new ListNode(0);
        ListNode* node = newNode;
        ListNode* curr = head->next;
        while(curr!=NULL){
            int sum=0;
            while(curr->val!=NULL){
                sum=sum+curr->val;
                curr=curr->next;
            }
            curr=curr->next;
            ListNode* temp = new ListNode(sum);
            node->next = temp;
            node=node->next;
        }
        ListNode* result=newNode->next;
        delete newNode;
        return result;
    }
};
